detectable/src/test/java/com/synopsys/integration/detectable/detectables/packagist/functional/PackagistDetectableTest.java

```diff
@@ -0,0 +1,137 @@
+            "           \"version\": \"v0.8.1\",",
+            "           \"source\": {",
+            "               \"type\": \"git\",",
+            "               \"url\": \"https://github.com/graphp/algorithms.git\",",
+            "               \"reference\": \"81db4049c35730767ec8f97fb5c4844234b86cef\"",
+            "           },",
+            "           \"require\": {",
+            "               \"clue/graph\": \"~0.9.0|~0.8.0\",",
+            "               \"php\": \">=5.3\"",
+            "           },",
+            "           \"require-dev\": {",
+            "               \"psr/log\": \"~1.0\",",
+            "               \"symfony/event-dispatcher\": \"~2.1\",",
+            "               \"symfony/phpunit-bridge\": \"~2.7\"",
+            "           }",
+            "       },",
+            "   ],",
+            "   \"packages-dev\": []",
+            "}"
+        );
+
+        addFile(Paths.get("composer.json"),
+            "{",
+            "   \"name\": \"clue/graph-composer\",",
+            "   \"version\": \"1.0.0\",",
+            "   \"keywords\": [\"dependency graph\", \"visualize dependencies\", \"visualize composer\"],",
+            "   \"license\": \"MIT\",",
+            "   \"require\": {",
+            "       \"php\": \"^5.3.6 || ^7.0\",",
+            "       \"clue/graph\": \"^0.9.0\",",
+            "       \"jms/composer-deps-analyzer\": \"0.1.*\",",
+            "       \"symfony/console\": \"^2.1 || ^3.0\",",
+            "       \"graphp/graphviz\": \"^0.2.0\"",
+            "   },",
+            "   \"autoload\": {",
+            "       \"psr-0\": { \"Clue\\\\GraphComposer\": \"src/\" }",
+            "   },",
+            "   \"bin\": [ \"bin/graph-composer\" ],",
+            "   \"extra\": {",
+            "       \"phar\": {",
+            "           \"bundler\": \"composer\"",
+            "       }",
+            "   },",
+            "   \"config\": {",
+            "       \"platform\": {",
+            "           \"php\": \"5.3.6\"",
+            "       }",
+            "   }",
+            "}"
+        );
+    }
+
+    @NotNull
+    @Override
+    public Detectable create(@NotNull final DetectableEnvironment detectableEnvironment) {
+        return detectableFactory.createComposerDetectable(detectableEnvironment, new ComposerLockDetectableOptions(true));
+    }
+
+    @Override
+    public void assertExtraction(@NotNull final Extraction extraction) {
+        Assertions.assertNotEquals(0, extraction.getCodeLocations().size(), "A code location should have been generated.");
+
+        final NameVersionGraphAssert graphAssert = new NameVersionGraphAssert(Forge.PACKAGIST, extraction.getCodeLocations().get(0).getDependencyGraph());
+
+        graphAssert.hasRootSize(1);
+        graphAssert.hasRootDependency("clue/graph", "v0.9.0");
+        graphAssert.hasParentChildRelationship("graphp/algorithms", "v0.8.1", "clue/graph", "v0.9.0");
```

Comment:
Seems like very few asserts for such a large input file.

Reply:
This one's input is a little noisy because packagist requires a composer.lock and a composer.json.  I can trim down some of the unused info, but there's nothing else to assert on the graph that is produced.